src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -28,15 +28,37 @@ const (
 )
 
 var (
-	log = golog.LoggerFor("flashlight.server")
+	log               = golog.LoggerFor("flashlight.server")
+	frontingProviders = map[string]func(*http.Request) bool{
+		// WARNING: If you add a provider here, keep in mind that Go's http
+		// library normalizes all header names so the first letter of every
+		// dash-separated "word" is uppercase while all others are lowercase.
+		// Also, try and check more than one header to lean on the safe side.
+		"cloudflare": func(req *http.Request) bool {
+			return hasHeader(req, "Cf-Connecting-Ip") || hasHeader(req, "Cf-Ipcountry") || hasHeader(req, "Cf-Ray") || hasHeader(req, "Cf-Visitor")
+		},
+		"cloudfront": func(req *http.Request) bool {
+			return hasHeader(req, "X-Amz-Cf-Id") || headerMatches(req, "User-Agent", "Amazon Cloudfront")
+		},
+	}
```

Comment:
> Most Lantern clients will be old for a while. If we don't support old clients, blocking our current front would leave out the vast majority of our users. I think if that was actually OK then adding new fronts wouldn't be a priority right now.

I wasn't suggesting that we block the current front.  The way that I would do it is to have a default front (the current) and use a different front if the new header is received.  That way, new clients can benefit from the additional fronts, while old clients can continue to operate as they have been.  Gaining access to additional fronts might be a good reason to upgrade your client, which is something that we want to encourage anyway.

The thing that doesn't feel right about the auto-detection functionality to me is that it relies on headers that weren't meant to be used for this purpose.  Even sticking with the current fronts, do we have any assurance that these headers are included 100% of the time and that they won't change without notice?  Or what if we end up tweaking some setting that affects the headers somehow?  The impact if the auto-detection fails is pretty severe, as we'll potentially be returning incorrect hosts which will basically make the service unusable.

> there's some desirable simplicity in leaving the client out of this

I actually think it would be simpler to use our own header rather than to reverse engineer the headers used by CDNs.  It's a technique that will work for all CDNs and overall seems less likely to break since we define and control the protocol.

All that said, I'm not putting my foot down, just making a suggestion - if you still think that autodetecting is the way to go, I won't stand in your way.